tests/testutils/goroutineassistant.go

```diff
@@ -18,23 +18,25 @@ import (
 	"fmt"
 	"sync"
 	"testing"
+
+	"github.com/coreos/rkt/Godeps/_workspace/src/github.com/steveeJ/gexpect"
 )
 
 type GoroutineAssistant struct {
-	s  chan string
+	s  chan error
 	wg sync.WaitGroup
 	t  *testing.T
 }
 
 func NewGoroutineAssistant(t *testing.T) *GoroutineAssistant {
 	return &GoroutineAssistant{
-		s: make(chan string),
+		s: make(chan error),
 		t: t,
 	}
 }
 
 func (a *GoroutineAssistant) Fatalf(s string, args ...interface{}) {
-	a.s <- fmt.Sprintf(s, args...)
+	a.s <- fmt.Errorf(s, args...)
```

Comment:
I'd rather have it thought out with unbuffered channel to take advantage of the synchronizing nature of these. This will force us to think about the written test a little more and improve the quality. Buffered channels lead to fire&forget runs, and we'd need another mechanism to synchronize/stop failed tests(routines).